### PR TITLE
migration to elm-0.18

### PR DESCRIPTION
**Pull request by aische**
I did not touch the tests folder, so there might still be some work to be done.

### CODE REVIEW

**Comment by arturopala** [comment]
I will test and release it tomorrow.

File: example/OptionalExample.elm [hunk 1]
```diff
@@ -67,12 +67,12 @@ streetNameOfAddress =
 
 regionOfPlace : Optional Place String
 regionOfPlace =
-    addressOfPlace `Monocle.Optional.compose` regionOfAddress
+    Monocle.Optional.compose addressOfPlace regionOfAddress
```
Comment (HappyAndHarmless) [inline example/OptionalExample.elm]: Should the arguments here be flipped? To the effect of:

```
addressOfPlace
|> Monocle.Optional.compose regionOfAddress
```

Or, for obvious naming:

```
addressOfPlace
|> Monocle.Optional.andThen regionOfAddress
```

File: src/Monocle/Optional.elm [hunk 2]
```diff
@@ -159,11 +159,11 @@ zip left right =
             left.getOption a
                 |> (\ma ->
                         Maybe.andThen
-                            ma
                             (\c ->
                                 right.getOption b
                                     |> Maybe.map (\d -> ( c, d ))
                             )
+                            ma
                    )
```
Comment (HappyAndHarmless) [inline src/Monocle/Optional.elm]: The lambda here can be dissolved.

**Comment by aische** [comment]
Right, the lambda is not necessary, removed it.

If we want to flip the arguments of compose, we have to do it in Lens, Iso, Optional and Prism. Do you think it's a good idea? How about adding a new composeBackwards function? (I think we should not call it 'andThen' because it has a different type signature than all other andThens).

**Comment by aische** [comment]
Thank you!

**Pull request merged**